docs/source/advanced/sagemaker-pipelines.rst

```diff
@@ -0,0 +1,303 @@
+* Understanding of graph neural networks and the `GraphStorm framework <https://graphstorm.readthedocs.io/en/latest/index.html>`_.
+
+Setting Up Your Environment
+---------------------------
+
+To work with GraphStorm SageMaker pipelines, you'll need the GraphStorm source code
+and a Python environment with the SageMaker SDK and AWS SDK (boto3) installed.
+
+1. First, clone the GraphStorm repository and navigate to the pipeline directory:
+
+.. code-block:: bash
+
+   git clone https://github.com/awslabs/graphstorm.git
+   cd graphstorm/sagemaker/pipeline
+
+2. Install the required Python packages:
+
+.. code-block:: bash
+
+   pip install sagemaker boto3
+
+Creating Your First Pipeline
+--------------------------
+
+GraphStorm provides Python scripts to help you create and manage SageMaker pipelines. The main tools are:
+
+* ``create_sm_pipeline.py``: Creates or updates pipeline definitions
+* ``pipeline_parameters.py``: Manages pipeline configuration
+* ``execute_sm_pipeline.py``: Runs pipeline executions
+
+Here's an example of how to create a basic pipeline that includes graph construction, training, and inference:
+
+.. code-block:: bash
+
+    python create_sm_pipeline.py \
+        --graph-construction-config-filename my_gconstruct_config.json \
+        --graph-name my-graph \
+        --graphstorm-pytorch-cpu-image-url 123456789012.dkr.ecr.us-west-2.amazonaws.com/graphstorm:sagemaker-cpu \
+        --input-data-s3 s3://input-bucket/data \
+        --instance-count 2 \
+        --jobs-to-run gconstruct train inference \
+        --output-prefix s3://output-bucket/results \
+        --pipeline-name my-graphstorm-pipeline \
+        --region us-west-2 \
+        --role arn:aws:iam::123456789012:role/SageMakerExecutionRole \
+        --train-inference-task node_classification \
+        --train-yaml-s3 s3://config-bucket/train.yaml
+
+This command sets up a pipeline with three main stages:
+
+1. Graph construction using the configuration in ``my_gconstruct_config.json``
+2. Model training using the settings in ``train.yaml``
+3. Inference using the trained model
+
+The pipeline will use 2 instances (specified by ``--instance-count``) for distributed training and inference.
+
+You'll need to provide:
+
+* A SageMaker execution role (``--role``) with appropriate permissions
+* A GraphStorm Docker image (``--graphstorm-pytorch-cpu-image-url``) for running the tasks
+* S3 locations for your input data and where to store results
+
+Running Pipeline Executions
+-------------------------
+
+Once you've created a pipeline, you can execute it using the ``execute_sm_pipeline.py`` script:
+
+.. code-block:: bash
+
+    python execute_sm_pipeline.py \
+        --pipeline-name my-graphstorm-pipeline \
+        --region us-west-2
+
+You can override default parameters during execution to customize the run:
+
+.. code-block:: bash
+
+    python execute_sm_pipeline.py \
+        --pipeline-name my-graphstorm-pipeline \
+        --region us-west-2 \
```

Comment:
Can we change the execution region?

Reply:
We can't change the execution region, but when I was writing the execution script I wasn't able to find a consistent way to derive it from the user environment so I made it a required arg instead.

We could look further into this and try to guess the execution region (every pipeline is unique per account/region combination)

Reply:
Can we add a hint in the doc highlighting this restriction, i.e., once the region is set for a pipeline, user can only run tasks on that region?